### ISSUE TITLE
RestAPI doc is still using operationId and we do not have a solid way to guarantee they are the same

### ISSUE THREAD

**Issue by ArcturusZhang** [Azure/typespec-azure#1701]
The docs team is using `operationId` to identify operations, and they decided to stick with swaggers.
The downstream SDK generators (at least dotnet SDK) want to provide extra information that our user could use to look up through the REST API docs. The solution that dotnet SDK provides is to write the `operationId` into xml docs.
TCGC did not pass through this `operationId`, therefore the downstream SDK generators need a concrete `operationId` instead of constructing an `operationId` again. But because of the existence of various name changing decorators, it is impossible for the SDK generators to calculate a correct `operationId`.
Similar reason that TCGC and the autorest emitter that emits the swagger calculates the `operationId` independently, therefore I think it is also problematic in this solution, because in independent calculation, there is no guarantee that they always get the same result which leads to an `operationId` in swagger to mismatch the `operationId`s in SDKs.

So either, we could convince docs team to use a different identifier for operations so that we could use in the downstream SDK generators.
Or TCGC and autorest could share a piece of code to calculate the operationIds and pass through this to SDK generators to guarantee the operationId for the same operation is always the same.

**Comment by allenjzhang** [comment]
Having different `operation_id` is definitely not ideal. As autorest depends on TCGC, it may make sense to have TCGC to expose operation_id and let autorest uses that. 

There is still a risk that TCGC one differs from what autorest was generating, but we can run against spec repo to see the diff and correct the logic to be consistent.

**Comment by allenjzhang** [comment]
So the conclusion is for the immediate solution: 
- instead using client name to generate operation id, we will use TypeSpec name
- scan repo for anything impacted and add explicit operation id override.

This may pending on getting RPSaaSMaster green & breaking change board review. @ArcturusZhang , any timeline pressure here?

**Comment by ArcturusZhang** [comment]
> So the conclusion is for the immediate solution:
> 
> * instead using client name to generate operation id, we will use TypeSpec name
> * scan repo for anything impacted and add explicit operation id override.
> 
> This may pending on getting RPSaaSMaster green & breaking change board review. [<img alt="" width="16" height="16" src="https://avatars.githubusercontent.com/u/10554446?s=64&amp;u=251e41d5f93f70258c825332ff9d605735aa8ef2&amp;v=4">@ArcturusZhang](https://github.com/ArcturusZhang) , any timeline pressure here?

No. This issue has been there for a long time therefore its impact is relatively small since it only impacts docs of .Net SDKs
